torch/utils/data/dataset.py

```diff
@@ -82,12 +87,26 @@ def register_function(cls, function_name, function):
         cls.functions[function_name] = function
 
     @classmethod
-    def register_datapipe_as_function(cls, function_name, cls_to_register):
+    def register_datapipe_as_function(cls, function_name, cls_to_register, enable_df_api_tracing=False):
         if function_name in cls.functions:
+            print(cls.functions)
             raise Exception("Unable to add DataPipe function name {} as it is already taken".format(function_name))
 
         def class_function(cls, source_dp, *args, **kwargs):
-            return cls(source_dp, *args, **kwargs)
+            result_pipe = cls(source_dp, *args, **kwargs)
+            if isinstance(result_pipe, Dataset):
+                if enable_df_api_tracing or isinstance(source_dp, DFIterDataPipe) or getattr(result_pipe, '_dp_cast_to_df', False):
+                    if function_name not in UNTRACABLE_DATAFRAME_PIPES:
+                        result_pipe = result_pipe.trace_as_dataframe()
+
+                if getattr(result_pipe, '_dp_nesting_depth', None) is None:
+                    result_pipe._dp_nesting_depth = getattr(source_dp, '_dp_nesting_depth', None)
+
+                if getattr(result_pipe, '_dp_contains_dataframe', None) is None:
+                    result_pipe._dp_contains_dataframe = getattr(source_dp, '_dp_contains_dataframe', None)
```

Comment:
Could we add comments toward attribute of `_dp_cast_to_df`, `_dp_nesting_depth` and `_dp_contains_dataframe`?